src/oscar/views/generic.py

```diff
@@ -27,7 +28,13 @@ def post(self, request, *args, **kwargs):
                 return self.response
             else:
                 messages.error(request, _("Invalid form submission"))
-        return super().post(request, *args, **kwargs)
+        try:
+            f = super().post
+        except AttributeError:
+            raise SuspiciousOperation(
```

Comment:
`messages.error(request, _("Invalid form submission"))` is used above.

happy to change it to that, but it seems `SuspiciousOperation` is actually more appropriate for both, and oscar currently never uses it.

Reply:
I prefer to reserve `SuspiciousOperation` for things that are very likely to be malicious, rather than things that can be accidental (e.g., a bad URL generated in a template), because it results in an empty 400 response to the client (and because it gets logged as a likely security issue, which I don't think this would be).

The docstring for the mixin says it is only usable with `DetailView` so I think we should just return `super().get()` if no action or post handler is found (with the "Invalid form submission" message added to the response).